### ISSUE TITLE
Build dockerfile is not working

### ISSUE THREAD

**Issue by eliassal** [Microsoft/sql-server-samples#165]
Hi, I am struggling this morning to run the dockerfiles suggested here but no single one is working. After downloading the windows core image,They are all failing on the line to downalod the sqlexpress with the following error

> 
> Step 9/11 : RUN powershell -Command (New-Object System.Net.WebClient).DownloadFile('%sql_express_download_url%', 'sqlexpress.exe') && /sqlexpress.exe /qs /x:setup && /setup/setup.exe /q /A
> CTION=Install /INSTANCENAME=SQLEXPRESS /FEATURES=SQLEngine /UPDATEENABLED=0 /SQLSVCACCOUNT="NT AUTHORITY\System" /SQLSYSADMINACCOUNTS="BUILTIN\ADMINISTRATORS" /TCPENABLED=1 /NPENABLED=0 /I
> ACCEPTSQLSERVERLICENSETERMS && del /F /Q sqlexpress.exe && rd /q /s setup
>  ---> Running in cd5cab6a2b72
> Exception calling "DownloadFile" with "2" argument(s): "The remote server
> returned an error: (404) Not Found."
> At line:1 char:1
> + (New-Object System.Net.WebClient).DownloadFile('https://go.microsoft. ...
> + ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
>     + CategoryInfo          : NotSpecified: (:) [], MethodInvocationException
>     + FullyQualifiedErrorId : WebException
> 
> The command 'cmd /S /C powershell -Command (New-Object System.Net.WebClient).DownloadFile('%sql_express_download_url%', 'sqlexpress.exe') && /sqlexpress.exe /qs /x:setup && /setup/setup.ex
> e /q /ACTION=Install /INSTANCENAME=SQLEXPRESS /FEATURES=SQLEngine /UPDATEENABLED=0 /SQLSVCACCOUNT="NT AUTHORITY\System" /SQLSYSADMINACCOUNTS="BUILTIN\ADMINISTRATORS" /TCPENABLED=1 /NPENABL
> ED=0 /IACCEPTSQLSERVERLICENSETERMS && del /F /Q sqlexpress.exe && rd /q /s setup' returned a non-zero code: 1

**Comment by eliassal** [comment]
Forgot to mention that I used dockerfile for
mssql-server-2016-express-sp1-windows
mssql-server-2016-express-windows
as well as 2014 but all giving the same errors

Thanks for your help

**Comment by dalibormesaric** [comment]
I also have this problem. The sql_express_download_url is not working, and people have complained about this before https://connect.microsoft.com/SQLServer/feedback/details/2902847/sql-server-2016-express-download-fails

**Comment by kallie-b** [comment]
I'm running into this issue as well. Any progress solving this?

```
 ...docker build -t mssql-server-2016-express-sp1-windows .
Sending build context to Docker daemon 6.144 kB
Step 1/11 : FROM microsoft/windowsservercore
---> 7baaa272ad7d
Step 2/11 : MAINTAINER Perry Skountrianos
---> Running in b83129e7b7f8
---> 5d2c9e3dbdc7
Removing intermediate container b83129e7b7f8
Step 3/11 : ENV sql_express_download_url "https://go.microsoft.com/fwlink/?linkid=829176"
---> Running in 8a8cb1f7c9aa
---> 2d0fa89315c7
Removing intermediate container 8a8cb1f7c9aa
Step 4/11 : ENV sa_password _
---> Running in 9188703a0bd3
---> 0790210e0188
Removing intermediate container 9188703a0bd3
Step 5/11 : ENV attach_dbs "[]"
---> Running in b67ea73911e6
---> 639914c6098d
Removing intermediate container b67ea73911e6
Step 6/11 : ENV ACCEPT_EULA _
---> Running in 48612f40de87
---> 2bd320b04eac
Removing intermediate container 48612f40de87
Step 7/11 : COPY . /
---> ced628a3ac90
Removing intermediate container 664848611c6b
Step 8/11 : WORKDIR /
---> Running in ddef355b2059
---> 01ee64472362
Removing intermediate container ddef355b2059
Step 9/11 : RUN powershell -Command (New-Object System.Net.WebClient).DownloadFile('%sql_express_download_url%', 'sqlexpress.exe') && /sqlexpress.exe /qs /x:setup
/ACTION=Install /INSTANCENAME=SQLEXPRESS /FEATURES=SQLEngine /UPDATEENABLED=0 /SQLSVCACCOUNT="NT AUTHORITY\System" /SQLSYSADMINACCOUNTS="BUILTIN\ADMINISTRATORS" /
D=0 /IACCEPTSQLSERVERLICENSETERMS && del /F /Q sqlexpress.exe && rd /q /s setup
---> Running in 87da1c76a0bd
Exception calling "DownloadFile" with "2" argument(s): "The remote server
returned an error: (404) Not Found."
At line:1 char:1
+ (New-Object System.Net.WebClient).DownloadFile('https://go.microsoft. ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ CategoryInfo          : NotSpecified: (:) [], MethodInvocationException
+ FullyQualifiedErrorId : WebException

The command 'cmd /S /C powershell -Command (New-Object System.Net.WebClient).DownloadFile('%sql_express_download_url%', 'sqlexpress.exe') && /sqlexpress.exe /qs /x
.exe /q /ACTION=Install /INSTANCENAME=SQLEXPRESS /FEATURES=SQLEngine /UPDATEENABLED=0 /SQLSVCACCOUNT="NT AUTHORITY\System" /SQLSYSADMINACCOUNTS="BUILTIN\ADMINISTRA
PENABLED=0 /IACCEPTSQLSERVERLICENSETERMS && del /F /Q sqlexpress.exe && rd /q /s setup' returned a non-zero code: 1
```

**Comment by eliassal** [comment]
Thanks Kalli for the confirmation. However, I managed to create the sqlexpress container from scratch and I am able to access it in SSMS from the host machine but not remotely. 
On the host I can connect to 
F9FEFB7986C2\SQLEXPRESS
port 1433 is listening on the host, I checked from remote machines with telnet, no firewall issues.
Any idea how we can access the sql remotely?
Why I need this because Visual studio is not present on the hosty machine and need to connect my dev projects from the machine where I have VS 2015 installed
